src/commonMain/kotlin/sim/rotation/criteria/ResourceMissingGte.kt

```diff
@@ -15,7 +16,25 @@ class ResourceMissingGte(data: RotationRuleCriterion) : Criterion(Type.RESOURCE_
         null
     }
 
+    inline fun <reified T : Enum<T>> valueOf(type: String): T? {
+        return java.lang.Enum.valueOf(T::class.java, type)
```

Comment:
We cannot use Java types, as it would be non-portable to the JS build.  A companion method on `Resource` named `getResourceFromString(str: String): Resource?` or something like that would probably be best.

Reply:
enumValueOf is implemented in Kotlin for JS, changed it to that, should work. Do you still want it in an extra method in Resource?